### DISCUSSION TITLE
Windows support in OpenFL tutorials

### DISCUSSION THREAD

**Discussion by Maxime-Perret** [securefederatedai/openfederatedlearning#546] [Q&A]
Hello,

I've been experimenting with OpenFL trying to run different tutorials. Since supporting Windows was mentioned in #525, I am personally using Windows 11 Pro and ran into two problems which I believe are OS related in the different tutorials.

- For `DataLoader`, having `num_workers` any larger than `0` causes an issue [as such](https://stackoverflow.com/questions/71713719/runtimeerror-dataloader-worker-pids-15876-2756-exited-unexpectedly). Thus, for Windows users, having `num_workers = 0` would be necessary. As far as I know, out of the [interactive api folder](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api), this concerns the tutorials [Dogs vs Cats](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_DogsCats_ViT), [Histology](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_Histology), [Histology FedCurv](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_Histology_FedCurv), [Kvasir UNet](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_Kvasir_UNet), [Lightning MNIST GAN](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_Lightning_MNIST_GAN), [MVTec PatchSVDD](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_MVTec_PatchSVDD), [Market Re-ID](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_Market_Re-ID), [MedMNIST 2D](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_MedMNIST_2D), [MedMNIST 3D](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_MedMNIST_3D).

- Some of the arguments in the plans of the tutorials (in `workspace/plan/plan.yaml`) contain `!!`, like for example `criterion: !!python/object:torch.nn.modules.loss.CrossEntropyLoss`. When parsing the plan in [`federated/plan/plan.py`](https://github.com/intel/openfl/blob/develop/openfl/federated/plan/plan.py), line 42 uses `safe_load` which together causes an issue [`could not determine a constructor for the tag`](https://death.andgravity.com/yaml-unknown-tag). Using the fix suggested [here](https://github.com/yaml/pyyaml/issues/266), I got it to work by replacing `safe_load(yaml_path.read_text())` by `load(yaml_path.read_text(), Loader=yaml.Loader)`.

Best

Edit: Updated my post and removed personal issues after fixing some of my problems with a clean reinstall of Python+Anaconda, and created issue #549 from what is left, so I guess it is more appropriate following up on the issue page.
